tests/TestCase.php

```diff
@@ -24,7 +23,25 @@
  */
 abstract class TestCase extends BaseTestCase
 {
-    use ExpectDeprecationTrait;
+    /** @var null|callable */
+    private $previouslyDefinedErrorHandler;
+
+    /** @var array<string> */
+    private array $expectedDeprecations = [];
+
+    /** @var array<string> */
+    private array $actualDeprecations = [];
+
+    protected function tearDown(): void
+    {
+        if (null !== $this->previouslyDefinedErrorHandler) {
+            foreach ($this->expectedDeprecations as $expectedDeprecation) {
+                self::assertContains($expectedDeprecation, $this->actualDeprecations);
+            }
```

Comment:
what if `count(actual) > count(expected)` ?
we could miss some exceptions. I suggest to check them too

Reply:
I will so that in separate PR, will look quite funny, I promise.